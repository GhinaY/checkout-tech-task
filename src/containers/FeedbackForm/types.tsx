import { FieldError, ValidationRule } from 'react-hook-form';
import { FeedbackSubmission } from '../../types';
import { TextFieldProps } from '@mui/material/TextField';

export type ControlledTextFieldProps = TextFieldProps  & {
  name: keyof FeedbackSubmission,
  fieldError: FieldError | undefined,
  helperText?: string,
  required?: boolean,
  maxLength?: number,
  additionalRules?: {[key: string]: ValidationRule },
};
