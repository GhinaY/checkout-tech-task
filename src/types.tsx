export type FeedbackSubmission = {
  name: string,
  email: string,
  rating: number,
  comment: string,
};

export type RatingsDistribution = {
  [key: number]: number,
};
