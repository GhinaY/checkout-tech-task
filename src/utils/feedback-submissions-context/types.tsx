import { FeedbackSubmission, RatingsDistribution } from '../../types'

export type FeedbackSubmissionsContextType = {
  addNewSubmission: (newSubmission: FeedbackSubmission) => void,
  ratingsDistribution: RatingsDistribution,
  submissions: FeedbackSubmission[],
}
