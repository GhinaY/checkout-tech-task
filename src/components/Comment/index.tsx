import Stack from '@mui/material/Stack';
import { Typography } from '@mui/material';

type Props = {
  email: string,
  comment: string,
};

function Comment({email, comment}: Props) {
  return (
    <Stack m={{ xs: 2, md: 4 }} useFlexGap>
      <Typography 
        variant='subtitle1' 
        sx={{ fontWeight: 'bold', color: 'text.secondary' }}
      >
        {email}
      </Typography>      
      <Typography>{comment}</Typography>      
    </Stack>
  );
}

export default Comment;
