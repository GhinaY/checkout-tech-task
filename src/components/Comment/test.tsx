/* eslint-disable react-refresh/only-export-components */
import { render } from '@testing-library/react'
import { FeedbackSubmission } from '../../types';
import Comment from './index';

const mockSubmission: FeedbackSubmission = {
  name: 'test name',
  email: 'test@test.com',
  rating: 5,
  comment: 'test comment',
}

describe('Comment', () => {
  test('Renders the component correctly', () => {  
    const { asFragment } = render(
      <Comment {...mockSubmission} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
