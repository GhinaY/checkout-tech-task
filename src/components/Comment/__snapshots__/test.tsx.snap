// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Comment > Renders the component correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-h4camh-MuiStack-root"
  >
    <h6
      class="MuiTypography-root MuiTypography-subtitle1 css-1hkc175-MuiTypography-root"
    >
      test@test.com
    </h6>
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      test comment
    </p>
  </div>
</DocumentFragment>
`;
